elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -229,6 +312,23 @@ public SecurityRealm get() throws StartException {
                             if (nameRewriter == null) {
                                 nameRewriter = NameRewriter.IDENTITY_REWRITER;
                             }
+                            KeyStore keyStore = keyStoreInjector.getOptionalValue();
+                            PrivateKey privateKey = null;
+                            PublicKey publicKey = null;
+                            if (keyStore != null) {
+                                    try {
+                                        keyStoreService = getModifiableKeyStoreService(keyStoreServiceRegistry, keyStoreName);
+                                        char[] keyPassword = getKeyStorePassword((KeyStoreService) keyStoreService);
+                                        privateKey = (PrivateKey) keyStore.getKey(keyPairAlias, keyPassword);
```

Comment:
Before attempting to get the private key and public key, would be good to check if the alias actually exists in the key store. If not, an appropriate exception should be thrown.